src/App.tsx

```diff
@@ -15,71 +16,93 @@ export const goods = [
+  selectedGood: string;
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends Component<{}, State> {
+  state = {
+    selectedGood: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  clearGood = () => {
+    this.setState({
+      selectedGood: '',
+    });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  setGood = (good: string) => {
+    this.setState({
+      selectedGood: good,
+    });
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  selectButton = (isSelected: boolean, good: string) => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+    isSelected
+      ? this.clearGood()
+      : this.setGood(good);
+  };
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  render() {
+    const { selectedGood } = this.state;
+
+    return (
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+      <main className="section container">
+        {selectedGood ? (
+          <h1 className="title is-flex is-align-items-center">
+            {`${selectedGood} is selected`}
 
-        <tr data-cy="Good">
-          <td>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
             <button
-              data-cy="AddButton"
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              className="delete ml-3"
+              onClick={this.clearGood}
+            />
+          </h1>
+        ) : (
+          <h1 className="title">No goods selected</h1>
+        )}
+
+        <table className="table">
+          <tbody>
+            {goods.map(good => {
+              const isSelected = selectedGood === good;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+              return (
+                <tr
+                  data-cy="Good"
+                  key={good}
+                  className={classNames(
+                    '',
+                    {
+                      'has-background-success-light': isSelected === true,
```

Comment:
This check would be enougth.
```suggestion
                      'has-background-success-light': isSelected,
```